spec/models/feedback_spec.rb

```diff
@@ -1,103 +1,37 @@
 require 'spec_helper'
 
-RSpec.describe Feedback, type: :model  do
+RSpec.describe Feedback, type: :model do
   subject(:feedback) { Fabricate.build(:feedback) }
 
-  let(:valid_feedback_token) { 'valid_feedback_token' }
-  let(:invalid_feedback_token) { 'invalid_feedback_token' }
-  let(:chapter) { Fabriate(:chapter) }
-
-  it { should respond_to(:request) }
-  it { should respond_to(:rating) }
-  it { should respond_to(:suggestions) }
-  it { should respond_to(:coach) }
-  it { should respond_to(:tutorial) }
-
   context 'validations' do
     context '#rating' do
-      it 'should not be blank' do
-        feedback = Fabricate.build(:feedback, rating: nil)
-
-        expect(feedback).to_not be_valid
-        expect(feedback).to have(1).error_on(:rating)
-      end
-
-      it 'should accept numbers from 1 to 5' do
-        (1..5).each do |rating|
-          feedback = Fabricate.build(:feedback, rating: rating)
-
-          expect(feedback).to be_valid
-        end
-      end
-
-      it 'should be invalid with number higher than 5' do
-        feedback = Fabricate.build(:feedback, rating: 6)
-
-        expect(feedback).to_not be_valid
-        expect(feedback).to have(1).error_on(:rating)
-      end
-
-      it 'should be invalid with number lower than 1' do
-        feedback = Fabricate.build(:feedback, rating: 0)
-
-        expect(feedback).to_not be_valid
-        expect(feedback).to have(1).error_on(:rating)
-      end
-
-      it 'should be numerical' do
-        feedback = Fabricate.build(:feedback, rating: 'alpha')
-
-        expect(feedback).to_not be_valid
-        expect(feedback).to have(1).error_on(:rating)
-      end
+      it { is_expected.to validate_presence_of(:rating) }
+      it { is_expected.to validate_inclusion_of(:rating).in_range(1..5).with_message(/can't be blank/) }
     end
 
-    context '#coach' do
-      it "accepts member with 'coach' role" do
-        feedback = Fabricate.build(:feedback, coach: Fabricate(:coach))
-
-        expect(feedback).to be_valid
-      end
-
-      it "is invalid with member with 'student' role" do
-        feedback = Fabricate.build(:feedback, coach: Fabricate(:student))
-
-        expect(feedback).to_not be_valid
-        expect(feedback).to have(1).error_on(:coach)
-      end
-
-      it "should display correct error message when member with 'student' role given" do
-        feedback = Fabricate.build(:feedback, coach: Fabricate(:student))
-
-        feedback.valid?
-
-        expect(feedback.errors.messages[:coach]).to include("Coach member doesn't have 'coach' role.")
-      end
-    end
+    it { is_expected.to validate_presence_of(:tutorial) }
   end
 
   context '#submit_feedback' do
     let(:feedback_request) { Fabricate(:feedback_request) }
 
-    let (:params) do
-      { rating: 4, coach: Fabricate(:coach), tutorial: Fabricate(:tutorial) }
-    end
+    let(:params) { rating: 4, coach: Fabricate(:coach), tutorial: Fabricate(:tutorial) }
```

Comment:
@notapatch I think it needs to contain a nested hash
`let(:params) {  { rating: 4, coach: Fabricate(:coach), tutorial: Fabricate(:tutorial) } }`

Reply:
That did indeed confuse me - I would say that nearly all of the submit_feedback code did leave me scratching my head. I put the Rubocop change in and backed away from the code slowly. 💣

Reply:
I think it's barely been touched since it was put in place as it just works. It could potentially do with a refactoring but then again there are so many aspects of the app that could be improved! 

Maybe we'll get to it when we put coach feeedback in place as it can probably be reused for that purpose.

Reply:
I would add it as a separate issue. Refactoring validations is relaxation between issues and I've been all too successful at not working at the cancelling issue. In that regards, I'm looking at refactoring workshop/_actions so that I can put cancelling in _without_ making the partial any more complex - I feel there's a slimmer, friendlier partial waiting to come out - it could also be delusion.

Reply:
@notapatch https://github.com/codebar/planner/issues/206